Core/Src/CAN/general_can.c

```diff
@@ -9,28 +9,50 @@
 #include "CAN/CAN_handler.h"
 
 //defined in main.c
-FDCAN_HandleTypeDef* hfdcan;
+FDCAN_HandleTypeDef* can2;
 
 FDCAN_TxHeaderTypeDef TxHeader;
 
+uint8_t RxData[8];
+FDCAN_RxHeaderTypeDef RxHeader;
+int16_t datageneral[4];
+uint32_t idgeneral;
 
 
+
+//função que inicializa a can geral, chamada em initializer.c
 void initialize_general_CAN(FDCAN_HandleTypeDef* can_ref) {
-	hfdcan = can_ref;
-	void CAN_general_receive_callback(FDCAN_HandleTypeDef* hfdcan);
-	initialize_CAN(hfdcan, CAN_general_receive_callback, &TxHeader);
+	can2 = can_ref;
+	void CAN_general_receive_callback(FDCAN_HandleTypeDef* hfdcan, uint32_t RxFifo0ITs);
```

Comment:
Só uma questão estética:
```suggestion
	void CAN_general_receive_callback(FDCAN_HandleTypeDef*, uint32_t);
```
Funciona do msm jeito e deixa menos verboso
(se vc preferir, pode definir a função em cima tbm, aí n precisa declarar ela antes assim. Eu pessoalmente prefiro do jeito q tá, pq a função de callback costuma ter lógica mais extensa, aí eu acho melhor deixar pro final do código)